cmd/build

```diff
@@ -1,51 +1,63 @@
 #!/bin/bash -e
+# The following are expected modes of builds
+# cmd/build <- Pull images referenced, then builds the ones that are missing 
+# cmd/build push <- Pushes images referenced in DOCKER_IMAGES_LIST to docker hub
+# cmd/build pull <- Pulls images referenced in DOCKER_IMAGES_LIST to docker hub
+# cmd/build build-all <- Builds all docker images 
 
 # Set this so docker build will fail even though piped through tee.
 set -o pipefail
 
 ROOT=$(dirname $0)/..
 
-build_root=build
-in_container=$DAQ_CONTAINER
-build_args=
-do_images=
-ext_msg=
 inline=
 copydirs="bin cmd misc faucet mininet docker daq subset"
-load_mark=/root/load_complete
+DOCKER_IMAGE_LIST=misc/docker_images.txt
+CHECK_EXISTING=y
 
 cd $ROOT
-DONT_MAKE_LOCAL=y
 source misc/config_base.sh
-
-if [ "$1" == skip ]; then
-    do_images=skip
-    shift
-elif [ "$1" == save ]; then
-    do_images=save
-    ext_msg=/save
-    shift
-elif [ "$1" == load ]; then
-    do_images=load
+if [ "$1" == push ]; then
+    cat $DOCKER_IMAGE_LIST | while read image hash other; do
+        found=$(docker images $image | awk '{print $3}' | grep $hash) || echo ""
+        if [ -z "$found" ]; then
+            echo $image:$hash not found locally...
+            continue
+        fi
+        echo Push $image:$hash to dockerhub...
+        TARGET="$image" VERSION="$hash" ./bin/docker_push  
```

Comment:
Why aren't these command line arguments?  This seems very odd.  Where else is docker_push used?